### PR TITLE
Update fix/clean up

### PR DESCRIPTION
**Pull request by NicholasTurner23**
# 🚀 Pull Request

## 📋 Description
**What does this PR do?**
- Add tests for analytics api components

**Why is this change needed?**
- Early detection of errors and catching edge cases
- To improve stability when changes are added

---

## 🔗 Related Issues
- [ ] OPS-229

---

## 🔄 Type of Change
- [ ] 🔧 Enhancement/improvement
- [ ] 📚 Documentation update
- [ ] ♻️ Refactor
- [ ] 🗑️ Removal/deprecation

## Summary by CodeRabbit

- New Features
  - None.
- Bug Fixes
  - Improved reliability of pagination cursors with better cache error handling and automatic expiration refresh on access.
  - Clearer errors for invalid or expired cursor tokens.
  - Safer data source lookup to prevent crashes when configuration keys are missing.
  - Graceful fallbacks when the cache is unavailable during data extraction.
- Tests
  - Added extensive tests for query building, pagination, cursors, data extraction, and cleaning.
  - Introduced an in-memory cache mock to isolate tests from external services.

### CODE REVIEW

File: src/analytics/api/utils/cursor_utils.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-import time
+import redis
+import logging
+
+logger = logging.getLogger(__name__)
+
+try:
+    RedisConnectionError = redis.ConnectionError
+except ImportError:
+    RedisConnectionError = ConnectionError
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply src/analytics/api/utils/cursor_utils.py:12]: Hi @NicholasTurner23 , the PR has already been merged. But I am noticing some powerful review comments here.

File: src/analytics/api/utils/cursor_utils.py [hunk 2]
```diff
@@ -17,45 +23,77 @@
...
         if not cursor_str:
             raise ValueError("Invalid or expired cursor token")
 
         # Reset the expiration time when the cursor is accessed
-        cache.set(cursor_key, cursor_str, timeout=CursorUtils.CURSOR_EXPIRATION)
+        try:
+            cache.set(cursor_key, cursor_str, timeout=CursorUtils.CURSOR_EXPIRATION)
+        except TypeError:
+            cache.set(cursor_key, cursor_str)
+            CursorUtils._set_cache_expiration(cursor_key)
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply src/analytics/api/utils/cursor_utils.py:97]: Hi @NicholasTurner23 , the PR has already been merged. But I am noticing some powerful review comments here.
